libraries/standard/mqtt/src/mqtt.c

```diff
@@ -99,6 +101,427 @@ static int32_t sendPacket( MQTTContext_t * pContext,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Convert a byte indicating a publish ack type to an #MQTTPubAckType_t.
+ *
+ * @param[in] packetType First byte of fixed header.
+ *
+ * @return Type of ack.
+ */
+static MQTTPubAckType_t getAckFromPacketType( uint8_t packetType )
+{
+    MQTTPubAckType_t ackType;
+    switch( packetType )
+    {
+        case MQTT_PACKET_TYPE_PUBACK:
+            ackType = MQTTPuback;
+            break;
+        case MQTT_PACKET_TYPE_PUBREC:
+            ackType = MQTTPubrec;
+            break;
+        case MQTT_PACKET_TYPE_PUBREL:
+            ackType = MQTTPubrel;
+            break;
+        case MQTT_PACKET_TYPE_PUBCOMP:
+            ackType = MQTTPubcomp;
+            break;
+        default:
+            /* Not an ack. */
+            assert( 0 );
+            break;
+    }
+    return ackType;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Dump a packet from the transport interface.
+ *
+ * @param[in] PContext MQTT Connection context.
+ * @param[in] remainingLength Remaining length of the packet to dump.
+ * @param[in] bytesAlreadyReceived Bytes previously read from the transport.
+ *
+ * @return #MQTTRecvFailed or #MQTTNoDataAvailable.
+ */
+static MQTTStatus_t dumpPacket( MQTTContext_t * const pContext,
+                                size_t remainingLength,
+                                int32_t bytesAlreadyReceived )
+{
+    MQTTStatus_t status = MQTTRecvFailed;
+    int32_t totalBytesReceived = bytesAlreadyReceived, bytesReceived = 0;
+    assert( pContext != NULL );
+    size_t bytesToReceive = pContext->networkBuffer.size;
+
+    while( totalBytesReceived < ( int32_t ) remainingLength )
+    {
+        /* Update number of bytes to receive. */
+        if( ( remainingLength - ( size_t ) totalBytesReceived ) < bytesToReceive )
+        {
+            bytesToReceive = remainingLength - ( size_t ) totalBytesReceived;
+        }
+
+        bytesReceived = pContext->transportInterface.recv( pContext->transportInterface.networkContext,
+                                                           pContext->networkBuffer.pBuffer,
+                                                           bytesToReceive );
+        if( bytesReceived != ( int32_t ) bytesToReceive )
+        {
+            /* Partial receive occurred while trying to dump packet. */
+            IotLogErrorWithArgs( "Partial receive while dumping packet."
+                                 "Received bytes=%d, Expected bytes=%u",
```

Comment:
I suggest avoiding whitespaces in the data section of the log (as shown in the examples of the logging standard) to avoid unnecessary characters. (We have been following the same in HTTP). Thus, I suggest changing it to: `ReceivedBytes=%s, ExpectedBytes=%d`

Reply:
My suggestion applies to all logs in the file for their data format

Reply:
Changed